### ISSUE TITLE
Vehicle Shop System

### ISSUE THREAD

**Issue by PatchRequest** [EveronLife/EveronLife#11]
**Summarize your proposal**
A generic system to build different kinds of vehicle shops

**Describe the solution you'd like**
The available vehicle and their price should be controlled by a config.
The player should see a preview and stats about the vehicle in the shops
In the future, an external connection for a dynamic system should be considered when implementing the system

**Comment by PhotoshopLOL** [comment]
1. This is the most performant but least preferable UX. Maybe implement this alongside 2 or 3 for more options for server owners?

2. On FiveM, they get around the "display car" limit by having a way to change the car displayed for each display slot. I agree that it looks really nice if you walk inside a dealership with display cars. +1 immersion but maybe prone to bug abuse since the vehicles are networked.

3. I think this is the most balanced way to handle a vehicle shop. Spawn the display car locally and let the player browse through the catalog as they please. Minimal performance hit server-side & multiple players can browse the vehicle shop at the same time. 

Spawning vehicles as soon as you purchase is the way to go. But sending it to the garage can help with performance esp. in high-pop. Maybe make it configurable also? (e.g. if there are no configured spawn points, vehicles will be sent to the garage)

**Comment by DrLuk** [comment]
I agree that the 3rd option seems the best perfomance and gameplay wise.

> Alternatively, vehicles could just be sent to the garage all the time to avoid the spawning problem altogether. But I personally think it's cool to drive off with the new vehicle where you buy it.

If possible, an option to choose would be awesome. 

> The vehicle owner has 60 seconds to drive off with the vehicle or else it will be sent to his garage 

Also important for normal garages, so nobody blocks the space for others when not taking action or disconnecting. I remember the chaos on a 100 player server after the restart on a saturday night, when everybody tried to get his car out asap...

**Comment by salutesh** [comment]
Could add the basic logics/configs/layouts based on this:
https://www.youtube.com/watch?v=8yzQq6hme30

**Comment by Friiize** [comment]
Is it possible to have 3D model view instead of picture so you could spin the car to see it fully before buyng  or just to have fun ? It will have some impact on performance but how much ?

**Comment by ThibaudetTevai** [comment]
> 2\. A shop that has the vehicles on-site standing there and you can walk around them and maybe sit inside. To purchase you interact with a shop sign next to it to choose a skin or whatever. This option looks nice but might require lots of space if a server has many vehicle types to offer.

It's the best for the experience . And a slot for a civilian work.
But if the server have more 50+ vechicle ... you can't put all the vehicle in same shop. Maybe create some shops for Truck , Car , Farming truck , etc ...

**Comment by bundoggity** [comment]
Going to give this one a go. Will update progress

**Comment by 12gabriel3** [comment]
I would like something like this: 
* Cars are spawned in
*  Player walks up to the car but don't have the option to get in
* Instead, on the driver seat, player has the option to hold F to buy
* After buying, player automatically gets in and can drive away

Nice features to have:
* Server owner can set a custom color palette for each vehicle spawn, each color has a chance to spawn defined by server owner
* This could also be done with vehicle models
* Tooltips saying how much space it has, max speed, engine power. Car characteristics in general
